packages/atomic/src/components/common/atomic-modal/atomic-modal.ts

```diff
@@ -0,0 +1,441 @@
+  }
+  
+  :host(.open) [part='backdrop'] {
+    @apply pointer-events-auto;
+  }
+  
+  :host(.open.dialog) [part='backdrop'] {
+    background-color: rgba(40, 40, 40, 0.8);
+  }
+  
+  :host(.dialog) [part='backdrop'] {
+    @apply grid;
+    @apply p-6;
+    transition: background-color 500ms ease-in-out;
+    grid-template-areas:
+      '. .     .'
+      '. modal .'
+      '. .     .';
+    grid-template-columns: 1fr min(30rem, 100%) 1fr;
+    grid-template-rows: 1fr auto 3fr;
+  }
+  
+  :host(.dialog) [part='container'] {
+    @apply rounded;
+    @apply shadow;
+  }
+  
+  :host(.dialog) [part='header-wrapper'] {
+    @apply px-6;
+    @apply py-4;
+  }
+  
+  :host(.dialog) [part='header'] {
+    @apply font-bold;
+  }
+  
+  :host(.dialog) [part='body-wrapper'] {
+    @apply p-6;
+  }
+  
+  :host(.dialog) [part='footer-wrapper'] {
+    @apply bg-neutral-light;
+    padding: 1rem 1.125rem;
+  }
+  
+  :host(.fullscreen) [part='container'] {
+    @apply absolute;
+    @apply inset-0;
+  }
+  
+  :host(.fullscreen) [part='header-wrapper'] {
+    @apply p-6;
+  }
+  
+  :host(.fullscreen) [part='body-wrapper'] {
+    @apply px-6;
+    @apply pt-8;
+    @apply pb-5;
+  }
+  
+  :host(.fullscreen) [part='footer-wrapper'] {
+    @apply shadow-t-lg;
+    @apply px-6;
+    @apply py-4;
+  }
+  `;
+
+  @state()
+  bindings!: AnyBindings;
+
+  @state()
+  error!: Error;
+
+  /**
+   * Whether to display the open and close animations over the entire page or the atomic-modal only.
+   */
+  @property({type: String, reflect: true}) boundary: 'page' | 'element' =
+    'page';
+
+  @property({type: Object, attribute: false}) close: () => void = () => {
+    this.isOpen = false;
+    return false;
+  };
+
+  /**
+   * The container to hide from the tabindex and accessibility DOM when the modal is closed.
+   */
+  @property({type: Object, attribute: false}) container?: HTMLElement;
+
+  @property({type: Boolean, reflect: true, converter: booleanConverter})
+  fullscreen = false;
+
+  @property({
+    type: Boolean,
+    reflect: true,
+    attribute: 'is-open',
+    converter: booleanConverter,
+  })
+  isOpen = false;
+
+  @property({type: Object, attribute: false}) onAnimationEnded: () => void =
+    () => {};
+
+  @property({type: Object, attribute: false}) scope?: HTMLElement;
+
+  @property({type: Object, attribute: false}) source?: HTMLElement;
+
+  private animatableContainer: Ref<HTMLElement> = createRef();
+  private currentWatchToggleOpenExecution = 0;
+  private focusTrap: Ref<HTMLAtomicFocusTrapElement> = createRef();
+  private headerId = randomID('atomic-modal-header-');
+  private shouldRender = false;
+  private updateBreakpoints = once(() => updateBreakpoints(this));
+
+  // Lifecycle methods
+
+  connectedCallback() {
+    super.connectedCallback();
+
+    document.body.addEventListener('keyup', this.handleCloseOnEscape);
+    document.body.addEventListener('touchmove', this.onWindowTouchMove, {
+      passive: false,
+    });
```

Comment:
Even tho this works find, personally, I like when the event management is handled by a reactive controller:
```ts
import {ReactiveController, ReactiveControllerHost} from 'lit';

class ModalKeyboardController implements ReactiveController {
  host: ReactiveControllerHost;
  
  constructor(host: ReactiveControllerHost, private onEscape: () => void) {
    this.host = host;
    host.addController(this);
  }
  
  hostConnected() {
    document.addEventListener('keyup', this.handleKeyup);
  }
  
  hostDisconnected() {
    document.removeEventListener('keyup', this.handleKeyup);
  }
  
  private handleKeyup = (e: KeyboardEvent) => {
    if (e.key?.toLowerCase() === 'escape') {
      this.onEscape();
    }
  };
}
```

Reply:
I'm not entirely against the suggestion, but I wonder if it just adds complexity rather than simplifying things.

I'd be entirely fine with adding a reactive controller to avoid code duplication (e.g., if the addEventListener logic was used by multiple components), but in this case I'm not so sure.